packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -104,41 +104,124 @@ export class AnchorSet {
+                if (element.parentIndex < src!.start) {
+                    numberBeforeMove = index;
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                } else if (element.parentIndex < src!.start + count) {
+                    numberToMove++;
+                } else {
                 // Fix indexes in src after moved items (subtract count).
+                    srcChildren[index].parentIndex -= count;
+                }
+            }
+            toMove = srcChildren.splice(numberBeforeMove, numberToMove);
+            if (srcChildren.length === 0) {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                srcParent!.afterEmptyField(src!.field);
+            }
                 toMove = [];
             } else {
                 toMove = [];
             }
+
+        if (dst === undefined) {
+            // Change is a delete.
+            // Moved items have already been un-parented, so just mark them as deleted.
+            for (const moved of toMove) {
+                assert(!moved.deleted, "PathNode must not be deleted");
+                moved.deleted = true;
+            }
+            return;
+        }
+
+        // Get dst (and set parent for moved items)
             let dstPath: PathNode | undefined;
             if (toMove.length > 0) {
                 // There are anchors which are getting moved,
                 // therefor the destination needs to be created if it does not yet exist.
-                dstPath = this.trackInner(dst);
+            dstPath = this.trackInner(dst.path);
+
+            // Update moved items for new parent.
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const offset = dst.start - src!.start;
+            for (const moved of toMove) {
+                moved.parentIndex += offset;
+                moved.parentPath = dstPath;
+                moved.parentField = dst.field;
+            }
             } else {
                 // There are no anchors to move,
                 // therefor we want to avoid creating the destination if it does not already exist.
-                dstPath = this.find(dst);
+            dstPath = this.find(dst.path);
                 if (dstPath !== undefined) {
                     // Since we need a remove ref below to handle the `toMove.length > 0` case above,
                     // add a ref here so that does not break this case.
                     dstPath.addRef();
                 }
             }
 
+        // Update dst
             if (dstPath !== undefined) {
-                // TODO: Fixup indexes for items in toMove (add dstIndex - start)
-                // TODO: Fixup indexes for items in dstPath after insertion (add count)
-                // TODO: insert toMove items into dstPath
+            // Update new parent to add moved children
+            let field = dstPath.children.get(dst.field);
```

Comment:
Looks like `field` could be const here if you don't do `field = toMove;` and just call `dstPath.children.set(dst.field, toMove);` instead.